units/add_new_metric_metadata.go

```diff
@@ -0,0 +1,73 @@
+package units
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/evergreen-ci/cedar"
+	"github.com/evergreen-ci/cedar/model"
+	"github.com/evergreen-ci/cedar/util"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/pkg/errors"
+)
+
+type PeriodicChangePointJob struct {
+	*job.Base `bson:"metadata" json:"metadata" yaml:"metadata"`
+	env       cedar.Environment
+	conf      *model.CedarConfig
+	queue     amboy.Queue
+}
+
+func init() {
+	registry.AddJobType("periodic-change-point-detection", func() amboy.Job { return makePeriodicChangePointJob() })
+}
+
+func makePeriodicChangePointJob() *PeriodicChangePointJob {
+	j := &PeriodicChangePointJob{
+		Base: &job.Base{
+			JobType: amboy.JobType{
+				Name:    "periodic-change-point-detection",
+				Version: 1,
+			},
+		},
+		env: cedar.GetEnvironment(),
```

Comment:
I think if the nil check of the environment at job-runtime populates the env, then it's safe to skip this.